Make the database kind queryable from AnyConnectionKind and AnyConnection

05storm26 opened the pull request.

abonander commented:
We already merged a PR to add a way to get the `AnyKind` of an `AnyPool`, it just hasn't been released yet: https://github.com/launchbadge/sqlx/blob/master/sqlx-core/src/pool/mod.rs#L348

I don't see the point of replacing the `AnyPool` typedef with a custom type, that could actually be a breaking change.

Adding `kind()` to `AnyConnection` seems like a good improvement though.

05storm26 commented:
@abonander  Oh, I have missed that. (I have tried to do that same thing using just `impl AnyPool` initially but didn't realize that `SharedPool` has the `connect_options` that can be used to get the actual `AnyKind`. So I rewrote `AnyPool` to be able to save the `AnyKind` from the `connect_options` during the initialization/creation. But yes it is unnecessary.) 

I removed the rewrite of `AnyPool` and just left the addition of `kind()` to `AnyConnectionKind` and `AnyConnection`.

05storm26 commented:
I have also added a kind getter to `PoolConnection<crate::any::Any>`

abonander commented on sqlx-core/src/pool/connection.rs:
```diff
@@ -129,6 +129,13 @@ impl<DB: Database> PoolConnection<DB> {
     }
 }
 
+#[cfg(feature = "any")]
+impl PoolConnection<crate::any::Any> {
+    pub fn kind(&self) -> crate::any::AnyKind {
+        self.pool.connect_options.kind()
+    }
+}
+
```
```suggestion
```

This isn't necessary because `PoolConnection` dereferences to `AnyConnection`, so calling `.kind()` will forward to `AnyConnection::kind()`.

05storm26 replied:
Done
